### PR TITLE
Bookstack grid view.

### PR DESCRIPTION
**Pull request by nileshdeepak**

### CODE REVIEW

**Comment by nileshdeepak** [comment]
Issues when changing screen size:
issue-181-pic-1
issue-181-pic-2

File: app/Http/Controllers/BookController.php [hunk 1]
```diff
@@ -41,6 +41,7 @@ public function index()
         $popular = $this->entityRepo->getPopular('book', 3, 0);
         $display = $this->currentUser->display;
         $this->setPageTitle('Books');
+
```
Comment (AbijeetP) [inline app/Http/Controllers/BookController.php]: Please undo this change.

File: app/Http/Controllers/BookController.php [hunk 2]
```diff
@@ -115,9 +123,17 @@ public function update(Request $request, $slug)
             'name' => 'required|string|max:255',
             'description' => 'string|max:1000'
         ]);
-        $book = $this->entityRepo->updateFromInput('book', $book, $request->all());
-        Activity::add($book, 'book_update', $book->id);
-        return redirect($book->getUrl());
+            
+            $input = $request->file('image')->getClientOriginalName();
+            echo $input;
```
Comment (AbijeetP) [inline app/Http/Controllers/BookController.php]: why the echo? Please remove it.

File: app/User.php [hunk 3]
```diff
@@ -22,7 +22,7 @@ class User extends Model implements AuthenticatableContract, CanResetPasswordCon
      * The attributes that are mass assignable.
      * @var array
      */
-    protected $fillable = ['name', 'email', 'image_id'];
+    protected $fillable = ['name', 'email', 'image_id', 'display']; //to write in user database
```
Comment (AbijeetP) [inline app/User.php]: what is this display property?

File: resources/views/users/edit.blade.php [hunk 4]
```diff
@@ -49,6 +49,15 @@
                             @endforeach
                         </select>
                     </div>
+                    <!--Select display type -->
+                    <div class="form-group">
+                        <label for="display">Type of view</label>
```
Comment (AbijeetP) [inline resources/views/users/edit.blade.php]: Name of the property needs to be changed.

File: app/Http/Controllers/BookController.php [hunk 5]
```diff
@@ -68,7 +69,14 @@ public function store(Request $request)
             'name' => 'required|string|max:255',
             'description' => 'string|max:1000'
         ]);
+        $image = $request->file('image');
+        $input = time().'-'.$image->getClientOriginalName();
```
Comment (AbijeetP) [inline app/Http/Controllers/BookController.php]: duplicate code - repeated in updateBook, please use a function to handle this.

File: app/Http/Controllers/BookController.php [hunk 6]
```diff
@@ -115,9 +123,17 @@ public function update(Request $request, $slug)
...
             'description' => 'string|max:1000'
         ]);
-        $book = $this->entityRepo->updateFromInput('book', $book, $request->all());
-        Activity::add($book, 'book_update', $book->id);
-        return redirect($book->getUrl());
+            
+            $input = $request->file('image')->getClientOriginalName();
+            echo $input;
+         $destinationPath = public_path('uploads/book/');
+         $request->file('image')->move($destinationPath, $input);
```
Comment (AbijeetP) [inline app/Http/Controllers/BookController.php]: `$input` will not be unique.

File: app/User.php (hunk 3 above)
Comment (AbijeetP) [inline app/User.php]: get rid of the comment.

Comment (AbijeetP) [inline app/User.php]: Migration query missing for - 

1. display property
2. book image path.

Please add this, and set the defaults.

File: resources/views/users/edit.blade.php [hunk 7]
```diff
@@ -49,6 +49,15 @@
...
+                        <select name="display" id="display">
+                            <option @if($user->display === 'grid') selected @endif value="grid">Grid</option>
+                            <option @if($user->display === 'list') selected @endif value="list">List</option>
+                        </select>
+                    </div>
+                    <!---->
```
Comment (AbijeetP) [inline resources/views/users/edit.blade.php]: no CSS changes are given.

File: resources/views/books/form.blade.php [hunk 8]
```diff
@@ -11,6 +11,10 @@
 </div>
 
 <div class="form-group">
+    <label for="image">Choose image</label>
```
Comment (AbijeetP) [inline resources/views/books/form.blade.php]: label has to be from language file.

File: resources/assets/sass/styles.scss [hunk 9]
```diff
@@ -274,5 +274,28 @@ $btt-size: 40px;
   }
 }
 
-
-
+.galleryItem {
```
Comment (AbijeetP) [inline resources/assets/sass/styles.scss]: needs to change, I'll discuss with you.

File: resources/assets/js/global.js [hunk 10]
```diff
@@ -153,6 +153,14 @@ $('[data-action="expand-entity-list-details"]').click(function() {
     $('.entity-list.compact').find('p').not('.empty-text').slideToggle(240);
 });
 
+// Toggle thumbnail::hide image and reduce grid size
+$(document).ready(function(){
+   $('[data-action="expand-thumbnail"]').click(function(){
+     $('.galleryItem').toggleClass("collapse");
```
Comment (AbijeetP) [inline resources/assets/js/global.js]: no need to select galleryItem again and again. "collapse" needs to be in single quote.

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 11]
```diff
@@ -17,6 +17,7 @@ public function up()
             $table->string('name');
             $table->string('email')->unique();
             $table->string('password', 60);
+            $table->string('books_display')->default('grid');
```
Comment (AbijeetP) [inline database/migrations/2014_10_12_000000_create_users_table.php]: please create a new migration file.
